test/model/tpp/test_deeptpp.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Third-party imports
+import pytest
+import mxnet as mx
+import numpy as np
+from mxnet import nd
+
+# First-party imports
+from gluonts.model.tpp.deeptpp._network import (
+    DeepTPPTrainingNetwork,
+    DeepTPPPredictionNetwork,
+)
+from gluonts.model.tpp.distribution import WeibullOutput
+
+
+def _allclose(a: nd.NDArray, b: nd.NDArray):
+    return np.allclose(a.asnumpy(), b.asnumpy(), atol=1e-6)
+
+
+TEST_CASES = [  # ia_times, marks, valid_length, num_marks, loglike
+    (
+        nd.array([[0.1, 0.2, 0.1, 0.12], [0.3, 0.15, 0.1, 0.12]]),
+        nd.array([[1, 2, 0, 2], [0, 0, 1, 2]]),
+        nd.array([3, 4]),
+        3,
+        nd.array([5.0570173, 6.6502876]),
+    ),
+    (
+        nd.array([[0.1, 0.2, 0.1, 0.12], [0.3, 0.15, 0.1, 0.12]]),
+        nd.array([[1, 2, 0, 2], [0, 0, 1, 2]]),
+        nd.array([0, 4]),
+        3,
+        nd.array([0.9517526, 6.6502876]),
+    ),
+    (
+        nd.array([[0.1, 0.2, 0.1, 0.12], [0.3, 0.15, 0.1, 0.12]]),
+        nd.array([[1, 2, 0, 2], [0, 0, 1, 2]]),
+        nd.array([0, 0]),
+        3,
+        nd.array([0.9517526, 0.9725293]),
+    ),
+    (
+        nd.array([[0.1, 0.2, 0.1, 0.12], [0.3, 0.15, 0.1, 0.12]]),
+        nd.array([[0, 0, 0, 0], [0, 0, 0, 0]]),
+        nd.array([3, 4]),
+        1,
+        nd.array([1.7608211, 2.2495322]),
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "ia_times, marks, valid_length, num_marks, loglike", TEST_CASES
+)
+def test_log_likelihood(ia_times, marks, valid_length, num_marks, loglike):
+    mx.rnd.seed(seed_state=1234)
+
+    model = DeepTPPTrainingNetwork(
+        num_marks=num_marks,
+        interval_length=2,
+        time_distr_output=WeibullOutput(),
+    )
+    model.initialize()
+
+    loglike_pred = model(nd.stack(ia_times, marks, axis=-1), valid_length)
+
+    assert loglike_pred.shape == (ia_times.shape[0],)
+    assert _allclose(loglike, loglike_pred)
```

Comment:
I see this assertion potentially creating problems in the future (say, if the rng method changes in mxnet): is this testing anything useful?

Reply:
I can imagine that some changes may happen inside `DeepTPPTrainingNetwork` or `Weibull`, and this test should hopefully catch if any mistakes are introduced in the log-likelihood computation. Still, I agree with you that this is not a very likely scenario, and the cons may outweigh the pros. @canerturkmen do you have any ideas about what other tests we could include to check the correctness of the `DeepTPPTrainingNetwork`? Or is it fine if we only check the shape of the loss?

Reply:
that's ok, we can leave this in now, and reconsider it whenever (if) it starts giving issues